src/server/snapshot.cc

```diff
@@ -285,12 +309,24 @@ unsigned SliceSnapshot::SerializePhysicalBucket(DbIndex db_index, PrimeTable::bu
     }
     error_code ec = tmp_serializer.FlushMem();
     CHECK(!ec && !sfile.val.empty());
-
-    dest_->Push(GetDbRecord(db_index, std::move(sfile.val), result));
+    PushFileToChannel(&sfile, db_index, result, !serializer_compression);
   }
   return result;
 }
 
+void SliceSnapshot::PushFileToChannel(io::StringFile* sfile, DbIndex db_index, unsigned num_records,
+                                      bool should_compress) {
+  string string_to_push = std::move(sfile->val);
+  if (should_compress && string_to_push.size() > 4096) {
```

Comment:
please introduce a named variable and maybe reduce its value to 256. 
also please check if compression makes sense at all.